enum.rb

```diff
@@ -0,0 +1,521 @@
+    num = nums.is_a?(Integer) && nums >= 1 ? nums : 0
+    unless block_given?
+      return to_enum :my_each_with_index,
+                     (print '=>', 'No block was given', "\n")
+    end
+    return nil, print('=>', 'Array is empty', "\n") if empty?
+
+    0.upto(length - num) do |index|
+      if num.is_a?(Integer) && num >= 1
+        aux = {}
+        index.upto(index + num - 1) do |i|
+          aux[self[i]] = yield self[i], i
+        end
+        print aux, "\n" unless aux.empty?
+      else
+        checked[self[index]] = yield self[index], index unless self[index].nil?
+      end
+    end
+    print checked unless checked.empty?
+  end
+
+  # rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+  # *********************************
+  # End of my_each_with_index method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_select method.
+  # *********************************
+  # rubocop:disable Metrics/MethodLength
+  def my_select
+    arr = []
+    unless block_given?
+      return to_enum :my_select, print('No block was given.', "\n", "\n")
+    end
+
+    if !empty?
+      0.upto(length - 1) do |index|
+        arr.push(self[index]) if yield(self[index])
+      end
+    else
+      print '=> ', 'Array is empty.', "\n", "\n"
+    end
+    print('=> ', arr, "\n", "\n") unless arr.empty?
+    arr
+  end
+
+  # rubocop: enable Metrics/MethodLength
+  # *********************************
+  # End of my_select method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_all method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality, Layout/EndAlignment
+  def my_all?(arg = nil)
+    temp = true
+    0.upto(length - 1) do |index|
+      temp &&= if block_given?
+                 yield(self[index])
+               else
+                 (arg === self[index])
+        end
+    end
+    temp
+  end
+
+  # rubocop: enable Style/CaseEquality, Layout/EndAlignment
+  # *********************************
+  # End of my_all? method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_any? method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality, Layout/EndAlignment
+  def my_any?(arg = nil)
+    temp = false
+    0.upto(length - 1) do |index|
+      temp ||= if block_given?
+                 yield(self[index])
+               else
+                 (arg === self[index])
+        end
+    end
+    temp
+  end
+
+  # rubocop: enable Style/CaseEquality, Layout/EndAlignment
+  # *********************************
+  # End of my_any? method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_none? method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality, Layout/EndAlignment
+  def my_none?(pat = nil)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
true_array = [nil, false, true, []] 
false_array = [nil, false, nil, false]
false_array.my_none? == true # should return true
true_array.my_none? == false # should return false
```

Reply:
Tests run satisfactorily. both equalities are true according to the original none? method.